app/controllers/admin/versions_controller.rb

```diff
@@ -0,0 +1,59 @@
+module Admin
+  class VersionsController < Admin::BaseController
+    skip_authorization_check
+    before_action :authorize_version, only: [:revert]
+
+    def index
+      conf_ids_for_organizer = current_user.is_admin? ? Conference.pluck(:id) : Conference.with_role(:organizer, current_user).pluck(:id)
+      @versions = PaperTrail::Version.where(["conference_id IN (?) OR item_type = 'User'", conf_ids_for_organizer]).
+                                                                  order('created_at DESC').paginate(page: params[:page], per_page: 20)
+    end
+
+    def revert
+      if params[:attribute]
+        if @version.event != 'update'
+          flash[:error] = 'Reverting a specific attribute is allowed only for update type changes'
+
+        elsif @version.item.nil?
+          flash[:error] = 'Reverting a specific attribute is not allowed for deleted objects'
+
+        elsif @version.changeset.reject{ |_, values| values[0].blank? && values[1].blank? }.keys.include?(params[:attribute])
+          if @version.item[params[:attribute]] == @version.changeset[params[:attribute]][0]
+            flash[:error] = 'The item is already in the state that you are trying to revert it back to'
+          else
+            @version.item[params[:attribute]] = @version.changeset[params[:attribute]][0]
+            @version.item.save!
+            flash[:notice] = 'The selected change was successfully reverted'
+          end
+
+        else
+          flash[:error] = 'Revert failed. Invalid attribute'
+        end
+
+      elsif @version.event != 'create'
+        @version.reify.save!
+        flash[:notice] = 'The selected change was successfully reverted'
+
+      elsif @version.event == 'create' && %w(Conference User).include?(@version.item_type)
+        flash[:error] = 'Reverting creation of a conference is not permitted'
+
+      elsif @version.event == 'create' && @version.item
+        # if @version represets a create event and is not currently deleted
+        @version.item.destroy
+        flash[:notice] = 'The selected change was successfully reverted'
+
+      else
+        flash[:error] = 'The item is already in the state that you are trying to revert it back to'
+      end
+      redirect_to admin_revision_history_path
+    end
+
+    private
+
+    def authorize_version
+      @version = PaperTrail::Version.find(params[:id])
+      return if current_user.has_role?(:organizer, Conference.find(@version.conference_id)) || current_user.is_admin
```

Comment:
You should also change that into an ability, so that you can check if the user can do something instead of checking if the user has a role.